### PR TITLE
fix:7102 - Added config to enable hot reload when project started with docker compose up

### LINKED ISSUE #7102: Hot reloading does not work when project is set using docker compose up.
### Description

When a developer sets up the project using docker. The project does not support hot reloading hence the dev has to stop the container and run it again every time a change is made.

I was able to identify a fix for this:

image

image

If this issue persists for other devs too, I can take this up and push a fix :)

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Harrish-Selvarajah**
… docker compose up

Closes #7102 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7644"></a>

### CODE REVIEW

**Review by Pradumnasaraf — COMMENTED** [review]
The docker part looks good. Over to @eddiejaoude for the Next.js part.

File: next.config.mjs [hunk 1]
```diff
@@ -37,6 +37,13 @@ const nextConfig = {
     locales: ["en"],
     defaultLocale: "en",
   },
+  webpackDevMiddleware: (config) => {
+    config.watchOptions = {
+      poll: 1000,
+      aggregateTimeout: 300,
+    };
+    return config;
+  },
```
Comment (eddiejaoude) [inline next.config.mjs:46]: Please share the docs for `webpackDevMiddleware`

Comment (Harrish-Selvarajah) [reply next.config.mjs:46]: Sorry for the late reply; the fix was sourced from stackoverflow. I will have a look at this over the weekend and get back :)

Comment (eddiejaoude) [reply next.config.mjs:46]: Thank you 👍 

I worry when things come from stackoverflow as they are often out of date.

Please check and let us know

**Comment by SaraJaoude** [comment]
Due to lack of activity I am closing this Pull Request.

File: next.config.mjs (hunk 1 above)
Comment (eddiejaoude) [reply next.config.mjs:46]: It has been 2 weeks with no response, so I will remove these changes and merge the docker changes

Comment (eddiejaoude) [inline next.config.mjs:46]:
```suggestion
```

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Pull request merged**